data/src/main/scala/cats/data/Or.scala

```diff
@@ -0,0 +1,511 @@
+    new Monoid[A Or B] {
+      def combine(a1: A Or B, a2: A Or B) =
+        a1 +++ a2
+      def empty =
+        ROr(Monoid[B].empty)
+    }
+}
+
+sealed abstract class OrInstances0 extends OrInstances1 {
+  implicit def OrEqual[A: Eq, B: Eq]: Eq[A Or B] =
+    new Eq[A Or B] {
+      def eqv(a1: A Or B, a2: A Or B) =
+        a1 === a2
+    }
+
+  /*
+  implicit def OrShow[A: Show, B: Show]: Show[A Or B] =
+    Show.show(_.show)
+    */
+
+  implicit def OrSemigroup[A: Semigroup, B: Semigroup]: Semigroup[A Or B] =
+    new Semigroup[A Or B] {
+      def combine(a1: A Or B, a2: A Or B) =
+        a1 +++ a2
+    }
+}
+
+sealed abstract class OrInstances1 extends OrInstances2 {
+  implicit def OrInstances1[L]: /*Traverse[L Or ?] with */ Monad[L Or ?] /*with Cozip[L Or ?] with Plus[L Or ?] with Optional[L Or ?] with MonadError[Or, L]*/ =
+    new /*Traverse[L Or ?] with */ Monad[L Or ?] /*with Cozip[L Or ?] with Plus[L Or ?] with Optional[L Or ?] with Monad[Or, L]*/ {
+      override def map[A, B](fa: L Or A)(f: A => B) =
+        fa map f
+
+      def flatMap[A, B](fa: L Or A)(f: A => L Or B) =
+        fa flatMap f
+
+      def pure[A](a: A) =
+        ROr(a)
+
+      def traverseImpl[G[_] : Applicative, A, B](fa: L Or A)(f: A => G[B]) =
+        fa.traverse(f)
+
+      /*
+      override def foldRight[A, B](fa: L Or A, z: Lazy[B])(f: (A, => B) => B) =
+        fa.foldRight(z)(f)
+      */
+
+      def cozip[A, B](x: L Or (A Or B)) =
+        x match {
+          case l @ -\/(_) => -\/(l)
+          case ROr(e) => e match {
+            case -\/(a) => -\/(ROr(a))
+            case b @ ROr(_) => ROr(b)
+          }
+        }
+
+      def plus[A](a: L Or A, b: => L Or A) =
+        a orElse b
+
+      def pextract[B, A](fa: L Or A): (L Or B) Or A = fa match {
+        case l@ -\/(_) => -\/(l)
+        case r@ ROr(_) => r
+      }
+
+      def raiseError[A](e: L): L Or A =
+        -\/(e)
+
+      def handleError[A](fa: L Or A)(f: L => L Or A): L Or A = fa match {
+        case -\/(e) => f(e)
+        case r => r
+      }
+    }
+}
+
+sealed abstract class OrInstances2 {
+  /*
+  implicit val OrInstances2 : Bitraverse[Or] = new Bitraverse[Or] {
+    override def bimap[A, B, C, D](fab: A Or B)
+                                  (f: A => C, g: B => D) = fab bimap (f, g)
+
+    def bitraverseImpl[G[_] : Applicative, A, B, C, D](fab: A Or B)
+                                                  (f: A => G[C], g: B => G[D]) =
+      fab.bitraverse(f, g)
+  }
+
+  implicit val OrAssociative: Associative[Or] = new Associative[Or] {
+    def reassociateLeft[A, B, C](f: Or[A, Or[B, C]]) =
+      f.fold(
+        a => Or.left(Or.left(a)),
+        _.fold(
+          b => Or.left(Or.right(b)),
+          Or.right(_)
+        )
+      )
+
+    def reassociateRight[A, B, C](f: Or[Or[A, B], C]) =
+      f.fold(
+        _.fold(
+          Or.left(_),
+          b => Or.right(Or.left(b))
+        ),
+        c => Or.right(Or.right(c))
+      )
+  }
+  */
```

Comment:
I see the following dichotomy of zombies:
1. Depends on types that don't exist yet.
2. Depends on types that won't exist.

I'd like to kill zombie type 2 and leave zombie type 1, if we can distinguish them.  But I'm not opposed to killing all the zombies and gradually adding things back as their dependencies are born.